spec/non_default_api_spec.rb

```diff
@@ -497,7 +497,9 @@ def app
   end
 
   context ':hide_format' do
-    before :all do
+    let(:doc_options) { Hash.new }
+    before do
+      stub_const('DOC_OPTIONS', doc_options)
```

Comment:
`shared_examples_for` can take an argument, and then instead of defining a class like `SimpleApiWithHiddenPaths`, define that dynamically: 

```ruby
Class.new(Grape::API) do
 ...
end
```

Does this help?